vite-project-kabinet/src/Content/PaginatedList.jsx

```diff
@@ -3,31 +3,76 @@ import { CardPreview } from "./CardPreview.jsx"
 import { useState } from "react";
 import { useEffect } from "react";
 import { PaginationNav } from "./PaginationNav.jsx";
-import { getPostsWithCategoryName } from "../constants/fetchUpdatedPosts"
 
 export const PaginatedList = ({itemsPerPage}) => {
     const [currentPage, setCurrentPage] = useState(1);
+    const [totalPages, setTotalPages] = useState(0)
     const [items, setItems] = useState([])
 
-    const totalPages = Math.ceil(items.length / itemsPerPage);
-
     const handlePageChange = (newPage) => {
         if (newPage >= 1 && newPage <= totalPages) {
             setCurrentPage(newPage);
         }
     };
 
+    const getPostsWithCategoryName = async (perPage, page) => {
```

Comment:
Jestli se nepletu, je tohle stejná funkce jako v `LoadedList`. Zvládneš vymyslet jak ji definovat jen na jednom místě a přepoužít?

Reply:
Myslela som, že to nejde, keďže vo funkcii nastavujem novú hodnotu premennej totalPages. Ale keď už viem, že je to možné, tak popremýšľam...

Reply:
Ty si můžeš jako argument funkce dát jinou funkci, kterou si zavoláš.

Reply:
Definovala som to na dvoch miestach, pretože tá funkcia je iná pre LoadedList (mobilní verze) a iná pre PaginatedList. Fetchujem tam iné posts. Teraz keď sú tie funkcie pod sebou, tak je to lepšie vidieť.